text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/MiniMessageParser.java

```diff
@@ -119,66 +117,75 @@ private void processTokens(final @NotNull StringBuilder sb, final @NotNull Strin
   }
 
   @NotNull Component parseFormat(final @NotNull String richMessage, final @NotNull ContextImpl context) {
-    final PlaceholderResolver combinedResolver = PlaceholderResolver.combining(context.placeholderResolver(), this.placeholderResolver);
+    final TagResolver combinedResolver = TagResolver.resolver(this.tagResolver, context.extraTags());
     final Consumer<String> debug = context.debugOutput();
     if (debug != null) {
       debug.accept("Beginning parsing message ");
       debug.accept(richMessage);
       debug.accept("\n");
     }
 
-    final Function<TagNode, Transformation> transformationFactory;
+    final TokenParser.TagProvider transformationFactory;
     if (debug != null) {
-      transformationFactory = node -> {
+      transformationFactory = (name, args, token) -> {
         try {
           debug.accept("Attempting to match node '");
-          debug.accept(node.name());
-          debug.accept("' at column ");
-          debug.accept(String.valueOf(node.token().startIndex()));
+          debug.accept(name);
+          debug.accept("'");
+          if (token != null) {
+            debug.accept(" at column ");
+            debug.accept(String.valueOf(token.startIndex()));
+          }
           debug.accept("\n");
 
-          final Transformation transformation = this.registry.get(this.sanitizePlaceholderName(node.name()), node.parts(), combinedResolver, context);
+          final @Nullable Tag transformation = combinedResolver.resolve(name, new ArgumentQueueImpl<>(context, args), context);
 
           if (transformation == null) {
             debug.accept("Could not match node '");
-            debug.accept(node.name());
+            debug.accept(name);
             debug.accept("'\n");
           } else {
             debug.accept("Successfully matched node '");
-            debug.accept(node.name());
-            debug.accept("' to transformation ");
-            debug.accept(transformation.examinableName());
+            debug.accept(name);
+            debug.accept("' to tag ");
+            debug.accept(transformation instanceof Examinable ? ((Examinable) transformation).examinableName() : transformation.getClass().getName());
```

Comment:
I'd rather not -- it would invite questions from users implementing tags on how to use `examination`, or potential incorrect usage if they don't override `toString()` themselves.